bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/AbstractHomekitAccessoryImpl.java

```diff
@@ -160,10 +161,14 @@ protected void unsubscribe(HomekitCharacteristicType characteristicType) {
         return null;
     }
 
+    @SuppressWarnings("unchecked")
     protected <T> T getAccessoryConfiguration(String key, T defaultValue) {
         final @Nullable Map<String, Object> configuration = accessory.getConfiguration();
-        if ((configuration != null) && (configuration.get(key) != null)) {
-            return (T) configuration.get(key);
+        if (configuration != null) {
+            Object value = configuration.get(key);
+            if (value != null && defaultValue != null && value.getClass().equals(defaultValue.getClass())) {
```

Comment:
Didn't think about that. Another way would be to check for `value != null` and catch `ClassCastException` instead of trying to check the class before.

Reply:
Well the ClassCastException wouldn't occur in this method but it would occur on the caller when it attempts to cast the generic result into what it thinks it should be.

Reply:
In fact, `defaultValue` could be annotated as `@NonNull` at least in the use-cases we have so far.

Reply:
if defaultValue is null we dont know to which class to cast to. So,  it must be not null. we can add @NotNull. 

alternative would be to create separate methods like getAccessoryConfigurationAsString, getAccessoryConfigurationAsInteger 

currently we have only these 2 cases: String and Integer and cannot think of anything else. 
it is configuration from metadata - so, it will be either String "" or number.
Switch MyFan "My Fan" { homekit="Fan.v2", alexa="Fan" **[ type="oscillating", speedSteps=3 ]** }

Reply:
I think the generic version is better. is theer any reason why we can't add null-annotations to these classes`?

Reply:
NonNull will not hurt here. added.